### PR TITLE
Demoing recipe json being put into glider

### PR DESCRIPTION
**Pull request by mlgi**
This one's a bit of a doozy of a PR. Essentially, I combined my storage.js with @Steven-Christensen 's glider/slider/carousel thingy, with some code reorganization. This way we can use our own recipe jsons, which we eventually(hopefully) will scrape automatically from website urls as mentioned in #77.

I also commented out a bunch of firebase scripts that weren't being used yet. Looks like we already have a firebase project going so that's nice to know for when we (if time permits) scale up to database and user creation. ~~I kinda got some experience on firebase auth and firestore?? Nothing for productionthough~~

## Created GliderRecipe.js
This is a custom component just heavily based off my RecipeCard.js from lab 6. Uses tag `<glider-recipe>`

### loadrecipes(list)
This function just loads a list of urls to recipe JSONs into our storage.recipeData for the rest of the app to use

### renderRecipesIntoGlider()
Just uses our custom `<glider-recipe>` component and loads its `<li>` into the glider carousel

### init()
Calls the above functions as appropriate and also initializes the glider **after** all the recipes have been placed into the glider. If they are placed after the glider is .mount() or whatever, they don't show up on the page. Might need to read into glidejs to see if a refresh is possible. I just took the code from Steven's glider initialization that was originally in index.html and placed it into our main.js.

Feel free to leave any questions as comments in this PR or just in our slack channel.

## Previewing
Since we don't have firebase deploying previews yet, you'll have to pull my branch and run a live server on your local machine for now. Let me know if there are any problems.

### CODE REVIEW

File: source/js/main.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
+import { storage } from "./storage.js";
```
Comment (TuckerFrandsen) [inline source/js/main.js:1]: Codacy has a fix for the issue: [There should be no space after '{'.](https://app.codacy.com/gh/cse110-fa21-group27/cse110-fa21-group27/pullRequest?prid=8437408)
```suggestion
import {storage } from "./storage.js";
```

File: source/js/main.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+window.addEventListener("DOMContentLoaded",init);
+
+const gliderConfig = {
+  focusAt: 'center', //this line seems to being nothing. i wanted it to maybe like, enable the non-translucence. or something like that
+  type: 'carousel',
+  perView: 3,
```
Comment (TuckerFrandsen) [inline source/js/main.js:8]: Codacy has a fix for the issue: [Expected object keys to be in ascending order. 'perView' should be before 'type'.](https://app.codacy.com/gh/cse110-fa21-group27/cse110-fa21-group27/pullRequest?prid=8437408)
```suggestion
  //this line seems to being nothing. i wanted it to maybe like, enable the non-translucence. or something like that
```

File: source/js/GliderRecipe.js [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+  }
+
+  set data(data) {
+
+    let cleanData = {};
+    /** 
+     * Making a more organized object for getting 
+     * the info from messy recipe json. Might not end up using all of it,
+     * I just yoinked this from my lab code
+     *
```
Comment (TuckerFrandsen) [inline source/js/GliderRecipe.js:14]: Codacy has a fix for the issue: [Trailing spaces not allowed.](https://app.codacy.com/gh/cse110-fa21-group27/cse110-fa21-group27/pullRequest?prid=8437408)
```suggestion
     *
```

File: source/js/GliderRecipe.js [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+  constructor(){
+    super();
+    let shadow = this.attachShadow({mode: 'open'});
+  }
+
+  set data(data) {
+
+    let cleanData = {};
+    /** 
+     * Making a more organized object for getting
```
Comment (TuckerFrandsen) [inline source/js/GliderRecipe.js:11]: Codacy has a fix for the issue: [Delete `·`](https://app.codacy.com/gh/cse110-fa21-group27/cse110-fa21-group27/pullRequest?prid=8437408)
```suggestion
     * Making a more organized object for getting
```

File: source/js/GliderRecipe.js [hunk 5]
```diff
@@ -0,0 +1,201 @@
+class GliderRecipe extends HTMLElement {
+  constructor(){
+    super();
+    let shadow = this.attachShadow({mode: 'open'});
```
Comment (TuckerFrandsen) [inline source/js/GliderRecipe.js:4]: Codacy has a fix for the issue: ['shadow' is never reassigned. Use 'const' instead.](https://app.codacy.com/gh/cse110-fa21-group27/cse110-fa21-group27/pullRequest?prid=8437408)
```suggestion
    const shadow = this.attachShadow({mode: 'open'});
```

File: source/js/RecipeCard.js [hunk 6]
```diff
@@ -0,0 +1,350 @@
+class RecipeCard extends HTMLElement {
+  constructor() {
+    // Part 1 Expose - TODO
+
+    // You'll want to attach the shadow DOM here
+    super();
+    let shadow = this.attachShadow({mode: 'open'});
```
Comment (TuckerFrandsen) [inline source/js/RecipeCard.js:7]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/cse110-fa21-group27/cse110-fa21-group27/pullRequest?prid=8437408)
```suggestion
    let shadow = this.attachShadow({mode: "open"});
```

File: source/js/RecipeCard.js [hunk 7]
```diff
@@ -0,0 +1,350 @@
+class RecipeCard extends HTMLElement {
+  constructor() {
```
Comment (TuckerFrandsen) [inline source/js/RecipeCard.js:2]: Codacy has a fix for the issue: [Missing space before function parentheses.](https://app.codacy.com/gh/cse110-fa21-group27/cse110-fa21-group27/pullRequest?prid=8437408)
```suggestion
  constructor () {
```

**Comment by mlgi** [comment]
Hi @TuckerFrandsen I'm gonna ignore these codacy suggestions for now c: Let me know if we've changed the settings on it or if I should update my Prettier config

**Pull request merged**